bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -476,6 +481,34 @@ public State getState(Widget w) {
     }
 
     /**
+     * Converts an item state to the type the widget supports (if possible)
+     *
+     * @param w - Widget in sitemap that shows the state
+     * @param s - State of the item
+     * @return the converted state or the original if conversion was not possible
+     */
+    private State convertState(Widget w, State s) {
+        State returnState = null;
+
+        if (w instanceof SwitchImpl) {
+            if (s instanceof PercentType) {
```

Comment:
you don't need to check for a specific type here - just use `s instanceof Convertible` so we keep the knowledge of what can be converted to what in a single place.